helix-core/src/test/java/org/apache/helix/manager/zk/TestZkBaseDataAccessor.java

```diff
@@ -239,7 +239,6 @@ public void testSyncCreateWithTTL() {
     Assert.assertNotNull(getRecord);
     Assert.assertEquals(getRecord.getSimpleFields().size(), 0);
 
-    System.clearProperty("zookeeper.extendedTypesEnabled");
```

Comment:
I only see removal on System.clearProperty but not System.setProperty. Is this property set by test cases and are not cleared?

Reply:
Some of the newly added tests need this property. But if we clear it in any of them, we run the risk of failing some other test that expects it to be there but finds it's not. Unless there's a way to avoid parallel test execution between different classes, I don't know how to get around this. Please let me know if you have any suggestions.

Reply:
I meant to ask when this "zookeeper.extendedTypesEnabled" property is ever set?

Reply:
Do you mean in tests or in real world? In real world, it's never set from Helix's/ZkClient's perspective. It would only be set on the ZK server, if it's intended for the ZK to support TTL and Container modes. In tests, whenever we want to test creating a ZNode in TTL or Container mode, this needs to be set. Otherwise, ZK server throws an UnimplementedException.

Reply:
Synced offline